### ISSUE TITLE
[Bug]: TPCC stock level sql parse error 

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#5038]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):0.0.6
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

mysql> SELECT count(*) AS low_stock FROM (
    ->     SELECT s_w_id, s_i_id, s_quantity
    ->         FROM bmsql_stock
    ->             WHERE s_w_id = 1 AND s_quantity < 1000 AND s_i_id IN (
    ->                 SELECT ol_i_id
    ->                     FROM bmsql_district
    ->                     JOIN bmsql_order_line ON ol_w_id = d_w_id
    ->                     AND ol_d_id = d_id
    ->                     AND ol_o_id >= d_next_o_id - 20
    ->                     AND ol_o_id < d_next_o_id
    ->                     WHERE d_w_id = 1 AND d_id = 1
    ->                     )
    ->                 );
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MatrixOne server version for the right syntax to use. syntax error at position 527; 
mysql>

**Comment by iamlinjunhong** [comment]
Not reproducible:
<img width="1166" alt="截屏2022-09-16 下午4 00 08" src="https://user-images.githubusercontent.com/49111204/190587890-317fd2d5-1cd9-4a1f-becf-645f25c76c35.png">

**Comment by aressu1985** [comment]
@iamlinjunhong

pls try commit: fc685762b89cd931bdac2c09b93f6bab4ceeb6d5

mysql> SELECT count(*) AS low_stock FROM (
    ->  SELECT s_w_id, s_i_id, s_quantity
    -> FROM bmsql_stock
    -> WHERE s_w_id = 1 AND s_quantity < 1000 AND s_i_id IN (
    -> SELECT ol_i_id
    -> FROM bmsql_district
    -> JOIN bmsql_order_line ON ol_w_id = d_w_id
    -> AND ol_d_id = d_id
    -> AND ol_o_id >= d_next_o_id - 20
    -> AND ol_o_id < d_next_o_id
    -> WHERE d_w_id = 1 AND d_id = 1
    -> )
    -> );
ERROR 20303 (HY000): SQL parser error: You have an error in your SQL syntax; check the manual that corresponds to your MatrixOne server version for the right syntax to use. syntax error at line 13 column 2 near "";
mysql>

**Comment by aressu1985** [comment]
SELECT count(*) AS low_stock FROM (
 SELECT s_w_id, s_i_id, s_quantity
FROM bmsql_stock
WHERE s_w_id = 1 AND s_quantity < 1000 AND s_i_id IN (
SELECT ol_i_id
FROM bmsql_district
JOIN bmsql_order_line ON ol_w_id = d_w_id
AND ol_d_id = d_id
AND ol_o_id >= d_next_o_id - 20
AND ol_o_id < d_next_o_id
WHERE d_w_id = 1 AND d_id = 1
)
);

create table bmsql_stock (
  s_w_id       integer       not null,
  s_i_id       integer       not null,
  s_quantity   integer,
  s_ytd        integer,
  s_order_cnt  integer,
  s_remote_cnt integer,
  s_data       varchar(50),
  s_dist_01    char(24),
  s_dist_02    char(24),
  s_dist_03    char(24),
  s_dist_04    char(24),
  s_dist_05    char(24),
  s_dist_06    char(24),
  s_dist_07    char(24),
  s_dist_08    char(24),
  s_dist_09    char(24),
  s_dist_10    char(24),
  primary key (s_w_id, s_i_id)
) PARTITION BY KEY(s_w_id);

create table bmsql_order_line (
  ol_w_id         integer   not null,
  ol_d_id         integer   not null,
  ol_o_id         integer   not null,
  ol_number       integer   not null,
  ol_i_id         integer   not null,
  ol_delivery_d   timestamp,
  ol_amount       decimal(6,2),
  ol_supply_w_id  integer,
  ol_quantity     integer,
  ol_dist_info    char(24),
  primary key (ol_w_id, ol_d_id, ol_o_id, ol_number)
) PARTITION BY KEY(ol_w_id);

create table bmsql_district (
  d_w_id       integer       not null,
  d_id         integer       not null,
  d_ytd        decimal(12,2),
  d_tax        decimal(4,4),
  d_next_o_id  integer,
  d_name       varchar(10),
  d_street_1   varchar(20),
  d_street_2   varchar(20),
  d_city       varchar(20),
  d_state      char(2),
  d_zip        char(9),
  primary key (d_w_id, d_id)
) PARTITION BY KEY(d_w_id);

**Comment by iamlinjunhong** [comment]
mysql also fail to execute:
<img width="1342" alt="截屏2022-10-31 下午4 55 49" src="https://user-images.githubusercontent.com/49111204/198969842-ff4842e0-5f10-4aab-b836-85328f3b0b38.png">

**Comment by iamlinjunhong** [comment]
you can modify as follow:
SELECT count(*) AS low_stock FROM (
SELECT s_w_id, s_i_id, s_quantity
FROM bmsql_stock
WHERE s_w_id = 1 AND s_quantity < 1000 AND s_i_id IN (
SELECT ol_i_id
FROM bmsql_district
JOIN bmsql_order_line ON ol_w_id = d_w_id
AND ol_d_id = d_id
AND ol_o_id >= d_next_o_id - 20
AND ol_o_id < d_next_o_id
WHERE d_w_id = 1 AND d_id = 1
)
) t;

**Comment by aressu1985** [comment]
> you can modify as follow: SELECT count(*) AS low_stock FROM ( SELECT s_w_id, s_i_id, s_quantity FROM bmsql_stock WHERE s_w_id = 1 AND s_quantity < 1000 AND s_i_id IN ( SELECT ol_i_id FROM bmsql_district JOIN bmsql_order_line ON ol_w_id = d_w_id AND ol_d_id = d_id AND ol_o_id >= d_next_o_id - 20 AND ol_o_id < d_next_o_id WHERE d_w_id = 1 AND d_id = 1 ) ) t;

I Know this way, but mo should support the origin sql

**Comment by aressu1985** [comment]
fixed: 1feeb04254c6fdab0ae0061ecc32349136d6ee10


mysql> 
mysql> 
mysql> SELECT count(*) AS low_stock FROM (
    -> SELECT s_w_id, s_i_id, s_quantity
    -> FROM bmsql_stock
    -> WHERE s_w_id = 1 AND s_quantity < 1000 AND s_i_id IN (
    -> SELECT ol_i_id
    -> FROM bmsql_district
    -> JOIN bmsql_order_line ON ol_w_id = d_w_id
    -> AND ol_d_id = d_id
    -> AND ol_o_id >= d_next_o_id - 20
    -> AND ol_o_id < d_next_o_id
    -> WHERE d_w_id = 1 AND d_id = 1
    -> )
    -> );
+-----------+
| low_stock |
+-----------+
|         0 |
+-----------+
1 row in set (0.01 sec)

mysql>
